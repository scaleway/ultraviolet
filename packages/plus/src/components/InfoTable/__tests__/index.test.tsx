import { shouldMatchEmotionSnapshot } from '@utils/test'
import { describe, it } from 'vitest'
import { InfoTable } from '..'

describe('InfoTable', () => {
  it('should work with default props', () =>
    shouldMatchEmotionSnapshot(
      <InfoTable>
        <InfoTable.Row templateColumns="repeat(3, 1fr)">
          <InfoTable.Cell title="title">cell</InfoTable.Cell>
          <InfoTable.Cell title="title">cell</InfoTable.Cell>
          <InfoTable.Cell title="title">cell</InfoTable.Cell>
        </InfoTable.Row>
        <InfoTable.Row templateColumns="2fr 1fr">
          <InfoTable.Cell title="title">cell with more text</InfoTable.Cell>
          <InfoTable.Cell title="title">cell</InfoTable.Cell>
        </InfoTable.Row>
        <InfoTable.Row templateColumns="1fr">
          <InfoTable.Cell title="title">cell</InfoTable.Cell>
        </InfoTable.Row>
        <InfoTable.Row templateColumns="repeat(3, 1fr)">
          <InfoTable.Cell title="title">
            cell cell cell cell cell cell cell cell cell cell cell cell cell
            cell cell cell cell cell cell cell cell cell cell cell cell cell
            cell cell cell cell
          </InfoTable.Cell>
          <InfoTable.Cell title="title">cell</InfoTable.Cell>
          <InfoTable.Cell title="title">cell</InfoTable.Cell>
        </InfoTable.Row>
      </InfoTable>,
    ))

  it('should work with multiLine', () =>
    shouldMatchEmotionSnapshot(
      <InfoTable>
        <InfoTable.Row templateColumns="repeat(3, 1fr)">
          <InfoTable.Cell title="title">cell</InfoTable.Cell>
          <InfoTable.Cell title="title">cell</InfoTable.Cell>
          <InfoTable.Cell title="title">cell</InfoTable.Cell>
        </InfoTable.Row>
        <InfoTable.Row templateColumns="2fr 1fr">
          <InfoTable.Cell title="title">cell with more text</InfoTable.Cell>
          <InfoTable.Cell title="title">cell</InfoTable.Cell>
        </InfoTable.Row>
        <InfoTable.Row templateColumns="1fr">
          <InfoTable.Cell title="title">cell</InfoTable.Cell>
        </InfoTable.Row>
        <InfoTable.Row templateColumns="repeat(3, 1fr)">
          <InfoTable.Cell title="title" multiline>
            cell cell cell cell cell cell cell cell cell cell cell cell cell
            cell cell cell cell cell cell cell cell cell cell cell cell cell
            cell cell cell cell
          </InfoTable.Cell>
          <InfoTable.Cell title="title">cell</InfoTable.Cell>
          <InfoTable.Cell title="title">cell</InfoTable.Cell>
        </InfoTable.Row>
      </InfoTable>,
    ))

  it('should work with width', () =>
    shouldMatchEmotionSnapshot(
      <InfoTable width="30%">
        <InfoTable.Row templateColumns="repeat(3, 1fr)">
          <InfoTable.Cell title="title">cell</InfoTable.Cell>
          <InfoTable.Cell title="title">cell</InfoTable.Cell>
          <InfoTable.Cell title="title">cell</InfoTable.Cell>
        </InfoTable.Row>
        <InfoTable.Row templateColumns="2fr 1fr">
          <InfoTable.Cell title="title">cell with more text</InfoTable.Cell>
          <InfoTable.Cell title="title">cell</InfoTable.Cell>
        </InfoTable.Row>
        <InfoTable.Row templateColumns="1fr">
          <InfoTable.Cell title="title">cell</InfoTable.Cell>
        </InfoTable.Row>
        <InfoTable.Row templateColumns="repeat(3, 1fr)">
          <InfoTable.Cell title="title">
            cell cell cell cell cell cell cell cell cell cell cell cell cell
            cell cell cell cell cell cell cell cell cell cell cell cell cell
            cell cell cell cell
          </InfoTable.Cell>
          <InfoTable.Cell title="title">cell</InfoTable.Cell>
          <InfoTable.Cell title="title">cell</InfoTable.Cell>
        </InfoTable.Row>
      </InfoTable>,
    ))
})
