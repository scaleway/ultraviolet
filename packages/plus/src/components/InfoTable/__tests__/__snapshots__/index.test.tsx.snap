// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`InfoTable > should work with default props 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 1rem;
  line-height: 1.5rem;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  margin: 0;
  width: 100%;
}

.emotion-0 .emotion-2 {
  width: 100%;
}

.emotion-3 {
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  -webkit-align-items: normal;
  -webkit-box-align: normal;
  -ms-flex-align: normal;
  align-items: normal;
  -webkit-box-pack: normal;
  -ms-flex-pack: normal;
  -webkit-justify-content: normal;
  justify-content: normal;
  border-bottom: 1px #d9dadd solid;
  padding-block: 1rem;
}

.emotion-3:first-of-type {
  padding-top: 0;
}

.emotion-3:last-of-type {
  padding-bottom: 0;
  border-bottom-color: transparent;
}

.emotion-3>*:not(:last-child) {
  padding-right: 1rem;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 0.25rem;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: normal;
  -webkit-box-align: normal;
  -ms-flex-align: normal;
  align-items: normal;
  -webkit-box-pack: normal;
  -ms-flex-pack: normal;
  -webkit-justify-content: normal;
  justify-content: normal;
  -webkit-box-flex-wrap: nowrap;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  width: 100%;
  min-width: 0;
}

.emotion-7 {
  font-weight: 500;
  color: #222638;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 0;
}

.emotion-9 {
  color: #727683;
  font-size: 0.875rem;
  font-family: Inter,sans-serif;
  font-weight: 500;
  letter-spacing: 0;
  line-height: 1.25rem;
  text-transform: none;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-11 {
  color: #3f4250;
  margin: 0;
  min-width: 0;
  width: 100%;
}

.emotion-14 {
  color: #3f4250;
  font-size: 1rem;
  font-family: Inter,sans-serif;
  font-weight: 400;
  letter-spacing: 0;
  line-height: 1.5rem;
  text-transform: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
  display: block;
  min-width: 0;
  width: 100%;
}

.emotion-14>* {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-right: 0.5rem;
}

.emotion-39 {
  display: grid;
  grid-template-columns: 2fr 1fr;
  -webkit-align-items: normal;
  -webkit-box-align: normal;
  -ms-flex-align: normal;
  align-items: normal;
  -webkit-box-pack: normal;
  -ms-flex-pack: normal;
  -webkit-justify-content: normal;
  justify-content: normal;
  border-bottom: 1px #d9dadd solid;
  padding-block: 1rem;
}

.emotion-39:first-of-type {
  padding-top: 0;
}

.emotion-39:last-of-type {
  padding-bottom: 0;
  border-bottom-color: transparent;
}

.emotion-39>*:not(:last-child) {
  padding-right: 1rem;
}

.emotion-64 {
  display: grid;
  grid-template-columns: 1fr;
  -webkit-align-items: normal;
  -webkit-box-align: normal;
  -ms-flex-align: normal;
  align-items: normal;
  -webkit-box-pack: normal;
  -ms-flex-pack: normal;
  -webkit-justify-content: normal;
  justify-content: normal;
  border-bottom: 1px #d9dadd solid;
  padding-block: 1rem;
}

.emotion-64:first-of-type {
  padding-top: 0;
}

.emotion-64:last-of-type {
  padding-bottom: 0;
  border-bottom-color: transparent;
}

.emotion-64>*:not(:last-child) {
  padding-right: 1rem;
}

<div
    data-testid="testing"
  >
    <dl
      class="emotion-0 emotion-1"
    >
      <div
        class="emotion-2 emotion-3 emotion-4"
      >
        <div
          class="emotion-5 emotion-6"
        >
          <dt
            class="emotion-7 emotion-8"
          >
            <p
              class="emotion-9 emotion-10"
            >
              title
            </p>
          </dt>
          <dd
            class="emotion-11 emotion-12"
          >
            <div
              class="emotion-13 emotion-14 emotion-10"
            >
              cell
            </div>
          </dd>
        </div>
        <div
          class="emotion-5 emotion-6"
        >
          <dt
            class="emotion-7 emotion-8"
          >
            <p
              class="emotion-9 emotion-10"
            >
              title
            </p>
          </dt>
          <dd
            class="emotion-11 emotion-12"
          >
            <div
              class="emotion-13 emotion-14 emotion-10"
            >
              cell
            </div>
          </dd>
        </div>
        <div
          class="emotion-5 emotion-6"
        >
          <dt
            class="emotion-7 emotion-8"
          >
            <p
              class="emotion-9 emotion-10"
            >
              title
            </p>
          </dt>
          <dd
            class="emotion-11 emotion-12"
          >
            <div
              class="emotion-13 emotion-14 emotion-10"
            >
              cell
            </div>
          </dd>
        </div>
      </div>
      <div
        class="emotion-2 emotion-39 emotion-4"
      >
        <div
          class="emotion-5 emotion-6"
        >
          <dt
            class="emotion-7 emotion-8"
          >
            <p
              class="emotion-9 emotion-10"
            >
              title
            </p>
          </dt>
          <dd
            class="emotion-11 emotion-12"
          >
            <div
              class="emotion-13 emotion-14 emotion-10"
            >
              cell with more text
            </div>
          </dd>
        </div>
        <div
          class="emotion-5 emotion-6"
        >
          <dt
            class="emotion-7 emotion-8"
          >
            <p
              class="emotion-9 emotion-10"
            >
              title
            </p>
          </dt>
          <dd
            class="emotion-11 emotion-12"
          >
            <div
              class="emotion-13 emotion-14 emotion-10"
            >
              cell
            </div>
          </dd>
        </div>
      </div>
      <div
        class="emotion-2 emotion-64 emotion-4"
      >
        <div
          class="emotion-5 emotion-6"
        >
          <dt
            class="emotion-7 emotion-8"
          >
            <p
              class="emotion-9 emotion-10"
            >
              title
            </p>
          </dt>
          <dd
            class="emotion-11 emotion-12"
          >
            <div
              class="emotion-13 emotion-14 emotion-10"
            >
              cell
            </div>
          </dd>
        </div>
      </div>
      <div
        class="emotion-2 emotion-3 emotion-4"
      >
        <div
          class="emotion-5 emotion-6"
        >
          <dt
            class="emotion-7 emotion-8"
          >
            <p
              class="emotion-9 emotion-10"
            >
              title
            </p>
          </dt>
          <dd
            class="emotion-11 emotion-12"
          >
            <div
              class="emotion-13 emotion-14 emotion-10"
            >
              cell cell cell cell cell cell cell cell cell cell cell cell cell cell cell cell cell cell cell cell cell cell cell cell cell cell cell cell cell cell
            </div>
          </dd>
        </div>
        <div
          class="emotion-5 emotion-6"
        >
          <dt
            class="emotion-7 emotion-8"
          >
            <p
              class="emotion-9 emotion-10"
            >
              title
            </p>
          </dt>
          <dd
            class="emotion-11 emotion-12"
          >
            <div
              class="emotion-13 emotion-14 emotion-10"
            >
              cell
            </div>
          </dd>
        </div>
        <div
          class="emotion-5 emotion-6"
        >
          <dt
            class="emotion-7 emotion-8"
          >
            <p
              class="emotion-9 emotion-10"
            >
              title
            </p>
          </dt>
          <dd
            class="emotion-11 emotion-12"
          >
            <div
              class="emotion-13 emotion-14 emotion-10"
            >
              cell
            </div>
          </dd>
        </div>
      </div>
    </dl>
  </div>
</DocumentFragment>
`;

exports[`InfoTable > should work with multiLine 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 1rem;
  line-height: 1.5rem;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  margin: 0;
  width: 100%;
}

.emotion-0 .emotion-2 {
  width: 100%;
}

.emotion-3 {
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  -webkit-align-items: normal;
  -webkit-box-align: normal;
  -ms-flex-align: normal;
  align-items: normal;
  -webkit-box-pack: normal;
  -ms-flex-pack: normal;
  -webkit-justify-content: normal;
  justify-content: normal;
  border-bottom: 1px #d9dadd solid;
  padding-block: 1rem;
}

.emotion-3:first-of-type {
  padding-top: 0;
}

.emotion-3:last-of-type {
  padding-bottom: 0;
  border-bottom-color: transparent;
}

.emotion-3>*:not(:last-child) {
  padding-right: 1rem;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 0.25rem;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: normal;
  -webkit-box-align: normal;
  -ms-flex-align: normal;
  align-items: normal;
  -webkit-box-pack: normal;
  -ms-flex-pack: normal;
  -webkit-justify-content: normal;
  justify-content: normal;
  -webkit-box-flex-wrap: nowrap;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  width: 100%;
  min-width: 0;
}

.emotion-7 {
  font-weight: 500;
  color: #222638;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 0;
}

.emotion-9 {
  color: #727683;
  font-size: 0.875rem;
  font-family: Inter,sans-serif;
  font-weight: 500;
  letter-spacing: 0;
  line-height: 1.25rem;
  text-transform: none;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-11 {
  color: #3f4250;
  margin: 0;
  min-width: 0;
  width: 100%;
}

.emotion-14 {
  color: #3f4250;
  font-size: 1rem;
  font-family: Inter,sans-serif;
  font-weight: 400;
  letter-spacing: 0;
  line-height: 1.5rem;
  text-transform: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
  display: block;
  min-width: 0;
  width: 100%;
}

.emotion-14>* {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-right: 0.5rem;
}

.emotion-39 {
  display: grid;
  grid-template-columns: 2fr 1fr;
  -webkit-align-items: normal;
  -webkit-box-align: normal;
  -ms-flex-align: normal;
  align-items: normal;
  -webkit-box-pack: normal;
  -ms-flex-pack: normal;
  -webkit-justify-content: normal;
  justify-content: normal;
  border-bottom: 1px #d9dadd solid;
  padding-block: 1rem;
}

.emotion-39:first-of-type {
  padding-top: 0;
}

.emotion-39:last-of-type {
  padding-bottom: 0;
  border-bottom-color: transparent;
}

.emotion-39>*:not(:last-child) {
  padding-right: 1rem;
}

.emotion-64 {
  display: grid;
  grid-template-columns: 1fr;
  -webkit-align-items: normal;
  -webkit-box-align: normal;
  -ms-flex-align: normal;
  align-items: normal;
  -webkit-box-pack: normal;
  -ms-flex-pack: normal;
  -webkit-justify-content: normal;
  justify-content: normal;
  border-bottom: 1px #d9dadd solid;
  padding-block: 1rem;
}

.emotion-64:first-of-type {
  padding-top: 0;
}

.emotion-64:last-of-type {
  padding-bottom: 0;
  border-bottom-color: transparent;
}

.emotion-64>*:not(:last-child) {
  padding-right: 1rem;
}

.emotion-89 {
  color: #3f4250;
  font-size: 1rem;
  font-family: Inter,sans-serif;
  font-weight: 400;
  letter-spacing: 0;
  line-height: 1.5rem;
  text-transform: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: block;
  min-width: 0;
  width: 100%;
}

.emotion-89>* {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-right: 0.5rem;
}

<div
    data-testid="testing"
  >
    <dl
      class="emotion-0 emotion-1"
    >
      <div
        class="emotion-2 emotion-3 emotion-4"
      >
        <div
          class="emotion-5 emotion-6"
        >
          <dt
            class="emotion-7 emotion-8"
          >
            <p
              class="emotion-9 emotion-10"
            >
              title
            </p>
          </dt>
          <dd
            class="emotion-11 emotion-12"
          >
            <div
              class="emotion-13 emotion-14 emotion-10"
            >
              cell
            </div>
          </dd>
        </div>
        <div
          class="emotion-5 emotion-6"
        >
          <dt
            class="emotion-7 emotion-8"
          >
            <p
              class="emotion-9 emotion-10"
            >
              title
            </p>
          </dt>
          <dd
            class="emotion-11 emotion-12"
          >
            <div
              class="emotion-13 emotion-14 emotion-10"
            >
              cell
            </div>
          </dd>
        </div>
        <div
          class="emotion-5 emotion-6"
        >
          <dt
            class="emotion-7 emotion-8"
          >
            <p
              class="emotion-9 emotion-10"
            >
              title
            </p>
          </dt>
          <dd
            class="emotion-11 emotion-12"
          >
            <div
              class="emotion-13 emotion-14 emotion-10"
            >
              cell
            </div>
          </dd>
        </div>
      </div>
      <div
        class="emotion-2 emotion-39 emotion-4"
      >
        <div
          class="emotion-5 emotion-6"
        >
          <dt
            class="emotion-7 emotion-8"
          >
            <p
              class="emotion-9 emotion-10"
            >
              title
            </p>
          </dt>
          <dd
            class="emotion-11 emotion-12"
          >
            <div
              class="emotion-13 emotion-14 emotion-10"
            >
              cell with more text
            </div>
          </dd>
        </div>
        <div
          class="emotion-5 emotion-6"
        >
          <dt
            class="emotion-7 emotion-8"
          >
            <p
              class="emotion-9 emotion-10"
            >
              title
            </p>
          </dt>
          <dd
            class="emotion-11 emotion-12"
          >
            <div
              class="emotion-13 emotion-14 emotion-10"
            >
              cell
            </div>
          </dd>
        </div>
      </div>
      <div
        class="emotion-2 emotion-64 emotion-4"
      >
        <div
          class="emotion-5 emotion-6"
        >
          <dt
            class="emotion-7 emotion-8"
          >
            <p
              class="emotion-9 emotion-10"
            >
              title
            </p>
          </dt>
          <dd
            class="emotion-11 emotion-12"
          >
            <div
              class="emotion-13 emotion-14 emotion-10"
            >
              cell
            </div>
          </dd>
        </div>
      </div>
      <div
        class="emotion-2 emotion-3 emotion-4"
      >
        <div
          class="emotion-5 emotion-6"
        >
          <dt
            class="emotion-7 emotion-8"
          >
            <p
              class="emotion-9 emotion-10"
            >
              title
            </p>
          </dt>
          <dd
            class="emotion-11 emotion-12"
          >
            <div
              class="emotion-13 emotion-89 emotion-10"
            >
              cell cell cell cell cell cell cell cell cell cell cell cell cell cell cell cell cell cell cell cell cell cell cell cell cell cell cell cell cell cell
            </div>
          </dd>
        </div>
        <div
          class="emotion-5 emotion-6"
        >
          <dt
            class="emotion-7 emotion-8"
          >
            <p
              class="emotion-9 emotion-10"
            >
              title
            </p>
          </dt>
          <dd
            class="emotion-11 emotion-12"
          >
            <div
              class="emotion-13 emotion-14 emotion-10"
            >
              cell
            </div>
          </dd>
        </div>
        <div
          class="emotion-5 emotion-6"
        >
          <dt
            class="emotion-7 emotion-8"
          >
            <p
              class="emotion-9 emotion-10"
            >
              title
            </p>
          </dt>
          <dd
            class="emotion-11 emotion-12"
          >
            <div
              class="emotion-13 emotion-14 emotion-10"
            >
              cell
            </div>
          </dd>
        </div>
      </div>
    </dl>
  </div>
</DocumentFragment>
`;

exports[`InfoTable > should work with width 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 1rem;
  line-height: 1.5rem;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  margin: 0;
  width: 100%;
}

.emotion-0 .emotion-2 {
  width: 30%;
}

.emotion-3 {
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  -webkit-align-items: normal;
  -webkit-box-align: normal;
  -ms-flex-align: normal;
  align-items: normal;
  -webkit-box-pack: normal;
  -ms-flex-pack: normal;
  -webkit-justify-content: normal;
  justify-content: normal;
  border-bottom: 1px #d9dadd solid;
  padding-block: 1rem;
}

.emotion-3:first-of-type {
  padding-top: 0;
}

.emotion-3:last-of-type {
  padding-bottom: 0;
  border-bottom-color: transparent;
}

.emotion-3>*:not(:last-child) {
  padding-right: 1rem;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 0.25rem;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: normal;
  -webkit-box-align: normal;
  -ms-flex-align: normal;
  align-items: normal;
  -webkit-box-pack: normal;
  -ms-flex-pack: normal;
  -webkit-justify-content: normal;
  justify-content: normal;
  -webkit-box-flex-wrap: nowrap;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  width: 100%;
  min-width: 0;
}

.emotion-7 {
  font-weight: 500;
  color: #222638;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 0;
}

.emotion-9 {
  color: #727683;
  font-size: 0.875rem;
  font-family: Inter,sans-serif;
  font-weight: 500;
  letter-spacing: 0;
  line-height: 1.25rem;
  text-transform: none;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-11 {
  color: #3f4250;
  margin: 0;
  min-width: 0;
  width: 100%;
}

.emotion-14 {
  color: #3f4250;
  font-size: 1rem;
  font-family: Inter,sans-serif;
  font-weight: 400;
  letter-spacing: 0;
  line-height: 1.5rem;
  text-transform: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
  display: block;
  min-width: 0;
  width: 100%;
}

.emotion-14>* {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-right: 0.5rem;
}

.emotion-39 {
  display: grid;
  grid-template-columns: 2fr 1fr;
  -webkit-align-items: normal;
  -webkit-box-align: normal;
  -ms-flex-align: normal;
  align-items: normal;
  -webkit-box-pack: normal;
  -ms-flex-pack: normal;
  -webkit-justify-content: normal;
  justify-content: normal;
  border-bottom: 1px #d9dadd solid;
  padding-block: 1rem;
}

.emotion-39:first-of-type {
  padding-top: 0;
}

.emotion-39:last-of-type {
  padding-bottom: 0;
  border-bottom-color: transparent;
}

.emotion-39>*:not(:last-child) {
  padding-right: 1rem;
}

.emotion-64 {
  display: grid;
  grid-template-columns: 1fr;
  -webkit-align-items: normal;
  -webkit-box-align: normal;
  -ms-flex-align: normal;
  align-items: normal;
  -webkit-box-pack: normal;
  -ms-flex-pack: normal;
  -webkit-justify-content: normal;
  justify-content: normal;
  border-bottom: 1px #d9dadd solid;
  padding-block: 1rem;
}

.emotion-64:first-of-type {
  padding-top: 0;
}

.emotion-64:last-of-type {
  padding-bottom: 0;
  border-bottom-color: transparent;
}

.emotion-64>*:not(:last-child) {
  padding-right: 1rem;
}

<div
    data-testid="testing"
  >
    <dl
      class="emotion-0 emotion-1"
    >
      <div
        class="emotion-2 emotion-3 emotion-4"
      >
        <div
          class="emotion-5 emotion-6"
        >
          <dt
            class="emotion-7 emotion-8"
          >
            <p
              class="emotion-9 emotion-10"
            >
              title
            </p>
          </dt>
          <dd
            class="emotion-11 emotion-12"
          >
            <div
              class="emotion-13 emotion-14 emotion-10"
            >
              cell
            </div>
          </dd>
        </div>
        <div
          class="emotion-5 emotion-6"
        >
          <dt
            class="emotion-7 emotion-8"
          >
            <p
              class="emotion-9 emotion-10"
            >
              title
            </p>
          </dt>
          <dd
            class="emotion-11 emotion-12"
          >
            <div
              class="emotion-13 emotion-14 emotion-10"
            >
              cell
            </div>
          </dd>
        </div>
        <div
          class="emotion-5 emotion-6"
        >
          <dt
            class="emotion-7 emotion-8"
          >
            <p
              class="emotion-9 emotion-10"
            >
              title
            </p>
          </dt>
          <dd
            class="emotion-11 emotion-12"
          >
            <div
              class="emotion-13 emotion-14 emotion-10"
            >
              cell
            </div>
          </dd>
        </div>
      </div>
      <div
        class="emotion-2 emotion-39 emotion-4"
      >
        <div
          class="emotion-5 emotion-6"
        >
          <dt
            class="emotion-7 emotion-8"
          >
            <p
              class="emotion-9 emotion-10"
            >
              title
            </p>
          </dt>
          <dd
            class="emotion-11 emotion-12"
          >
            <div
              class="emotion-13 emotion-14 emotion-10"
            >
              cell with more text
            </div>
          </dd>
        </div>
        <div
          class="emotion-5 emotion-6"
        >
          <dt
            class="emotion-7 emotion-8"
          >
            <p
              class="emotion-9 emotion-10"
            >
              title
            </p>
          </dt>
          <dd
            class="emotion-11 emotion-12"
          >
            <div
              class="emotion-13 emotion-14 emotion-10"
            >
              cell
            </div>
          </dd>
        </div>
      </div>
      <div
        class="emotion-2 emotion-64 emotion-4"
      >
        <div
          class="emotion-5 emotion-6"
        >
          <dt
            class="emotion-7 emotion-8"
          >
            <p
              class="emotion-9 emotion-10"
            >
              title
            </p>
          </dt>
          <dd
            class="emotion-11 emotion-12"
          >
            <div
              class="emotion-13 emotion-14 emotion-10"
            >
              cell
            </div>
          </dd>
        </div>
      </div>
      <div
        class="emotion-2 emotion-3 emotion-4"
      >
        <div
          class="emotion-5 emotion-6"
        >
          <dt
            class="emotion-7 emotion-8"
          >
            <p
              class="emotion-9 emotion-10"
            >
              title
            </p>
          </dt>
          <dd
            class="emotion-11 emotion-12"
          >
            <div
              class="emotion-13 emotion-14 emotion-10"
            >
              cell cell cell cell cell cell cell cell cell cell cell cell cell cell cell cell cell cell cell cell cell cell cell cell cell cell cell cell cell cell
            </div>
          </dd>
        </div>
        <div
          class="emotion-5 emotion-6"
        >
          <dt
            class="emotion-7 emotion-8"
          >
            <p
              class="emotion-9 emotion-10"
            >
              title
            </p>
          </dt>
          <dd
            class="emotion-11 emotion-12"
          >
            <div
              class="emotion-13 emotion-14 emotion-10"
            >
              cell
            </div>
          </dd>
        </div>
        <div
          class="emotion-5 emotion-6"
        >
          <dt
            class="emotion-7 emotion-8"
          >
            <p
              class="emotion-9 emotion-10"
            >
              title
            </p>
          </dt>
          <dd
            class="emotion-11 emotion-12"
          >
            <div
              class="emotion-13 emotion-14 emotion-10"
            >
              cell
            </div>
          </dd>
        </div>
      </div>
    </dl>
  </div>
</DocumentFragment>
`;
